src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -99,6 +99,26 @@ def update_image_pull_secrets(spec, image_pull_secrets):
     ]
 
 
+def update_volume_mounts(spec, volume_mounts: list):
+    if volume_mounts is []:
+        return None
```

Comment:
This can be removed, the `for` loop won't execute anything if the list is empty. Here an below.